### PR TITLE
Remove auth token and port openstack module_utils changes to v2 tree

### PR DESCRIPTION
**Pull request by emonty**

### CODE REVIEW

**Comment by emonty** [comment]
@bcoca we should pop this one in around the same time as the ansible-modules-extras one. I just read through the outstanding PRs on the old modules, and there is one feature someone wanted that's really good that will need this and the corresponding change there (which is already pushed up) After that, it's a piece of cake to support.

**Comment by bcoca** [comment]
my worry here is that this will break existing modules, even if deprecated they should still work

**Comment by emonty** [comment]
@bcoca the existing deprecated modules do not use this function, they use the other one.

File: lib/ansible/utils/module_docs_fragments/openstack.py [hunk 1]
```diff
@@ -69,12 +60,35 @@ class ModuleDocFragment(object):
       - How long should ansible wait for the requested resource.
     required: false
     default: 180
+  api_timeout:
+    description:
+      - How long should the socket layer wait before timing out for API calls.
+        If this is omitted, nothing will be passed to the requests library.
+    required: false
+  verify:
```
Comment (bcoca) [inline lib/ansible/utils/module_docs_fragments/openstack.py]: is this the same as "validate_certs" in other modules? if so I would prefer to standardize the naming to that

File: lib/ansible/utils/module_docs_fragments/openstack.py [hunk 2]
```diff
@@ -69,12 +60,35 @@ class ModuleDocFragment(object):
       - How long should ansible wait for the requested resource.
     required: false
     default: 180
+  api_timeout:
+    description:
+      - How long should the socket layer wait before timing out for API calls.
+        If this is omitted, nothing will be passed to the requests library.
+    required: false
```
Comment (bcoca) [inline lib/ansible/utils/module_docs_fragments/openstack.py:67]: non required fields should show default, even if just null

**Comment by bcoca** [comment]
aside from the 2 comments everything else looks good

**Pull request merged**